src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/PartialViewResultExecutor.cs

```diff
@@ -158,5 +160,47 @@ public virtual Task ExecuteAsync(ActionContext actionContext, IView view, Partia
                 viewResult.ContentType,
                 viewResult.StatusCode);
         }
+
+        private static string GetActionName(ActionContext context)
```

Comment:
Not sure I like this change mainly because ActionName is something which is calculated only once by the application model builder and should be reused rather than be calculated for every request.
I see your point about wanting to avoid downcasting to ControllerActionDescriptor.

My 2 cents: So considering the above maybe we can just leave `ActionName` property on `ActionDescriptor` as it is and avoid making an changes?